plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/procedure/FlushMetadataCacheProcedure.java

```diff
@@ -70,11 +72,13 @@ public class FlushMetadataCacheProcedure
     }
 
     private final Optional<CachingHiveMetastore> cachingHiveMetastore;
+    private final Optional<GlueCache> glueCache;
 
     @Inject
-    public FlushMetadataCacheProcedure(Optional<CachingHiveMetastore> cachingHiveMetastore)
+    public FlushMetadataCacheProcedure(Optional<CachingHiveMetastore> cachingHiveMetastore, Optional<GlueCache> glueCache)
```

Comment:
I suggest we add a single interface for this, say `MetastoreCacheControl` or `FlushMetastoreCache` or something.  Then this code could be generic again.